core/src/agent/state.rs

```diff
@@ -95,6 +95,12 @@ impl AgentState {
             .iter_type(&self.top_chain_header(), &entry.entry_type())
             .find(|h| h.entry_address() == &entry.address())
     }
+
+    pub fn get_header_for_entry_address(&self, entry_address: &Address) -> Option<ChainHeader> {
```

Comment:
`get_first_header_for_entry_address`?

there can be many headers for a single entry, even within a single chain

also i think we used to have this but i removed it because it boils down to applying a `find` to an `iter` that already exists

also `self.chain().iter(&self.top_chain_header())` feels wrong somehow :/

Reply:
Thanks for pointing this out, it was a real bug.

I introduced `iter_chain` to  `AgentState` for this purpose, which just iterates from the top header, what do you think?